src/libstore/globals.cc

```diff
@@ -114,7 +119,50 @@ std::vector<Path> getUserConfigFiles()
 
 unsigned int Settings::getDefaultCores()
 {
-    return std::max(1U, std::thread::hardware_concurrency());
+    unsigned int concurrency = std::max(1U, std::thread::hardware_concurrency());
+
+    #if __linux__
+    FILE *fp = fopen("/proc/mounts", "r");
+    if (!fp)
+        return concurrency;
+
+    Strings cgPathParts;
+
+    struct mntent *ent;
+    while ((ent = getmntent(fp))) {
+        std::string mountType, mountPath;
+
+        mountType = ent->mnt_type;
+        mountPath = ent->mnt_dir;
+
+        if (mountType == "cgroup2") {
+            cgPathParts.push_back(mountPath);
+            break;
+        }
+    }
+
+    fclose(fp);
+
+    if (cgPathParts.size() > 0 && pathExists("/proc/self/cgroup")) {
+        std::string currentCgroup = readFile("/proc/self/cgroup");
```

Comment:
I think this might blow-up if `/proc/self/cgroup` exists but is unreadable for whatever reason. Which is somewhat unlikely, but better be on the safe side

Reply:
Thank you for the suggestions. I've made some changes. How does it look now?